src/Raven.Server/NotificationCenter/Notifications/Details/RequestLatencyDetail.cs

```diff
@@ -1,82 +1,118 @@
+
             foreach (var key in RequestLatencies.Keys)
             {
                 var queue = RequestLatencies[key];
-                if (queue == null)
-                    continue;
-
                 var list = new DynamicJsonArray();
-                foreach (var details in queue)
+                if (queue != null)
                 {
-                    list.Add(details.ToJson());
+                    foreach (var details in queue)
+                    {
+                        list.Add(details.ToJson());
+                    }
                 }
 
-                dict[key] = list;
+                latencies[key] = list;
             }
 
-            return djv;
+            return new DynamicJsonValue
+            {
+                [nameof(RequestLatencies)] = latencies,
+                [Constants.Documents.Metadata.Key] = new DynamicJsonValue
+                {
+                    [Constants.Documents.Metadata.RavenClrType] = typeof(RequestLatencyDetail).AssemblyQualifiedName
+                }
+            };
         }
     }
 
-    public struct RequestLatencyInfo : IDynamicJsonValueConvertible
+    public class RequestLatencyInfo 
     {
-        public readonly string QueryString;
-        public readonly long Duration;
-        public readonly DateTime Date;
-        public readonly string Action;
+        public string Path;
+        public long Duration;
+        public DateTime Date;
+        public string Database;
+        public Dictionary<string,string> Parameters;
+        public long RequestContentLength;
 
-        public RequestLatencyInfo(string queryString, long duration, string action)
+        //for deserialization
+        protected RequestLatencyInfo()
         {
-            QueryString = queryString;
+            
+        }
+        
+        public RequestLatencyInfo(string path, long duration, string database, Parameters parameters, long requestContentLength)
+        {
+            Path = path;
             Duration = duration;
-            Action = action;
+            Database = database;
+            Parameters = parameters.ToDictionary(kvp => kvp.Key, kvp => Convert.ToString(kvp.Value));
+            RequestContentLength = requestContentLength;
             Date = DateTime.UtcNow;
         }
 
         public DynamicJsonValue ToJson()
         {
             return new DynamicJsonValue
             {
-                [nameof(QueryString)] = QueryString,
+                [nameof(Path)] = Path,
                 [nameof(Duration)] = Duration,
                 [nameof(Date)] = Date,
-                [nameof(Action)] = Action
+                [nameof(Database)] = Database,
+                [nameof(Parameters)] = Parameters.ToJson(),
+                [nameof(RequestContentLength)] = RequestContentLength,
+                [Constants.Documents.Metadata.Key] = new DynamicJsonValue
+                {
+                    [Constants.Documents.Metadata.RavenClrType] = typeof(RequestLatencyInfo).FullName
```

Comment:
is this needed?

Reply:
No, it ins't, neither is the `@metadata` section